core/src/main/scala/org/apache/spark/sql/rapids/tool/util/ToolsPlanGraph.scala

```diff
@@ -85,23 +34,30 @@ object PhotonSparkPlanGraphCluster {
  * Build a SparkPlanGraph from the root of a SparkPlan tree.
  */
 object ToolsPlanGraph {
-  // TODO: We should have a util to detect if the runtime is Databricks.
-  //      This can be achieved by checking for spark properties
-  //      spark.databricks.clusterUsageTags.clusterAllTags
-  private lazy val dbRuntimeReflection = DBReflectionContainer()
-
-  // By default call the Spark constructor. If this fails, we fall back to the DB constructor
-  def constructGraphNode(id: Long, name: String, desc: String,
-      metrics: collection.Seq[SQLPlanMetric], isPhotonNode: Boolean = false): SparkPlanGraphNode = {
-    try {
-      if (isPhotonNode) {
-        PhotonSparkPlanGraphNode(id, name, desc, metrics)
-      } else {
-        new SparkPlanGraphNode(id, name, desc, metrics)
-      }
-    } catch {
-      case _: java.lang.NoSuchMethodError =>
-        dbRuntimeReflection.constructNode(id, name, desc, metrics)
+  // Captures the API loaded at runtime if any.
+  var api: GraphReflectionAPI = _
+
+  // The actual code used to build the graph. If the API is not available, then fallback to the
+  // Spark default API.
+  private lazy val graphBuilder: SparkPlanInfo => SparkPlanGraph = {
+
+  GraphReflectionAPIHelper.api match {
+      case Some(_) =>
+        // set the api to the available one
+        api = GraphReflectionAPIHelper.api.get
```

Comment:
nit: why not use the value in `Some(_)` instead of `GraphReflectionAPIHelper.api.get`

Reply:
I think this code block is unrelated to the PR. It occurred due to merging with dev branch. I will revert the changes in this code block to avoid any confusion.